core/src/main/java/com/rim/IceField/Iceberg.java

```diff
@@ -99,9 +108,20 @@ public void setIsStable(boolean b) {
     }
 
     public void Add_currentPlayers(PlayerBase p) {
-        currentPlayers.add(p);
-        p.setCurrentIceberg(this);
-      //  System.out.println("Player" + p.ID + " was added to the iceberg number: " + num);
+        if(currentPlayers.size() < maxNumOfPlayers) {
+            currentPlayers.add(p);
+            p.setCurrentIceberg(this);
+            //  System.out.println("Player" + p.ID + " was added to the iceberg number: " + num);
+        }
+        else
+        {
+            System.out.println("Too many people on this iceberg! It has cracked.");
+            for(PlayerBase pl : currentPlayers)
+            {
```

Comment:
specifically, I do not understand why the players are falling

Reply:
They are falling when the number of players on the iceberg is bigger than it can hold an the iceberg breaks.